### PR TITLE
Merlin Learns to QMK

### PR DESCRIPTION
**Pull request by mechmerlin**
Move some of the more common things I'm doing in the layouts/community directory and putting it in my own userspace.

### CODE REVIEW

**Comment by drashna** [comment]
You may want to add `process_record_keymap` (or whatever name) here, so that you can specify macros "per keyboard". 

See: https://docs.qmk.fm/#/feature_userspace?id=consolidated-macros

(also, the KC_MAKE macro is nice)

**Comment by mechmerlin** [comment]
I haven't found a use case for macros yet actually. When I get there I'll add that in!

**Comment by drashna** [comment]
Okay. :)

File: users/mechmerlin/mechmerlin.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#ifndef USERSPACE
```
Comment (drashna) [inline users/mechmerlin/mechmerlin.h]: Probably should use `#pragma once` here, for the ifndef, define and endif.

**Comment by mechmerlin** [comment]
Done!

**Comment by drashna** [comment]
Thanks!

**Pull request merged**